apps/web/composables/useReadCookieBar/useReadCookieBar.ts

```diff
@@ -0,0 +1,153 @@
+import { toRefs } from '@vueuse/shared';
+import type { UseReadCookieBarState, UseReadCookieBarReturn } from './types';
+import { Cookie, CookieGroup, CookieGroupFromNuxtConfig } from 'cookie.config';
+import { ChangeVisibilityState, SetAllCookiesState, SetConsent, InitializeCookies } from './types';
+
+const checkIfScriptIsExternal = (scriptName: string): boolean => {
+  return scriptName.startsWith('http');
+};
+
+const fetchScripts = (scripts: string[]) => {
+  scripts.forEach((script: string) => {
+    try {
+      if (checkIfScriptIsExternal(script)) {
+        fetch(script, { method: 'GET', mode: 'no-cors', credentials: 'same-origin' })
+          .then((response) => response.text())
+          .then((text) => eval(text))
+          .catch(() => {
+            return;
+          });
+      } else {
+        eval(script);
+      }
+    } catch (error: any) {
+      console.error(error);
+    }
+  });
+};
+
+/**
+ * @description Composable for managing cookie consent bar.
+ * @returns UseReadCookieBarReturn
+ * @example
+ * ``` ts
+ * const {
+ *   data, visible, loading, changeVisibilityState, setConsent, initializeCookies, setAllCookiesState
+ * } = useReadCookieBar();
+ * ```
+ */
+export const useReadCookieBar: UseReadCookieBarReturn = () => {
+  const state = useState<UseReadCookieBarState>('useReadCookieBar', () => ({
+    data: {} as CookieGroupFromNuxtConfig,
+    visible: false,
+    loading: false,
+  }));
+
+  const runtimeConfig = useRuntimeConfig();
+  const initialCookies = runtimeConfig.public.cookieGroups as CookieGroupFromNuxtConfig;
+
+  const changeVisibilityState: ChangeVisibilityState = () => {
+    state.value.visible = !state.value.visible;
+  };
+
+  const loadThirdPartyScripts = (): void => {
+    if (!process.server) {
+      state.value.data.groups.forEach((cookieGroup: CookieGroup, groupIndex: number) => {
+        cookieGroup.cookies.forEach((cookie: Cookie, cookieIndex: number) => {
+          if (cookie.accepted) {
+            const scripts = initialCookies.groups[groupIndex].cookies?.[cookieIndex]?.script;
+
+            if (scripts && scripts.length > 0) {
+              fetchScripts(scripts);
+            }
+          }
+        });
+      });
+    }
+  };
+
+  /**
+   * @description Function for initializing cookies.
+   * @returns InitializeCookies
+   * @example
+   * ``` ts
+   * initializeCookies();
+   * ```
+   */
+  const initializeCookies: InitializeCookies = () => {
+    const cookies = JSON.parse(JSON.stringify(initialCookies));
+
+    const browserCookies = useCookie('consent-cookie');
+
+    cookies.groups.slice(1).forEach((group: CookieGroup) => {
+      group.cookies.forEach((cookie: Cookie) => {
+        cookie.accepted = !!browserCookies.value?.[group.name as any]?.[cookie.name as any] || false;
+      });
+
+      group.accepted = group.cookies.some((cookie: Cookie) => cookie.accepted);
+    });
+
+    state.value.data = cookies;
+
+    loadThirdPartyScripts();
```

Comment:
If the cookie is not set yet, this method should set the visibility to true so that the bar is shown when the user visits the site for the first time.